### PR TITLE
[beez3 & other] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #10399 .
#### Summary of Changes

Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')
#### Testing Instructions

Please do a code review or test all changed files.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on cdd4017ddf08288bf4ea84587fa25f2a839ed8b2<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on cdd4017ddf08288bf4ea84587fa25f2a839ed8b2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by zero-24** [comment]
Travis was failing on the sysinfo problem that was fixed upstream.

File: templates/beez3/html/mod_login/default.php [hunk 1]
```diff
@@ -17,9 +17,9 @@
 	<?php if ($params->get('greeting')) : ?>
 		<div class="login-greeting">
 		<?php if ($params->get('name') == 0) : ?>
-			<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('name'))); ?>
+			<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('name')), ENT_COMPAT, 'UTF-8'); ?>
```
Comment (andrepereiradasilva) [inline templates/beez3/html/mod_login/default.php]: Parentheses is in the wrong place.

File: templates/beez3/html/mod_login/default.php [hunk 2]
```diff
@@ -17,9 +17,9 @@
...
 		<?php else : ?>
-		 	<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('username'))); ?>
+		 	<?php echo JText::sprintf('MOD_LOGIN_HINAME', htmlspecialchars($user->get('username')), ENT_COMPAT, 'UTF-8'); ?>
```
Comment (andrepereiradasilva) [inline templates/beez3/html/mod_login/default.php]: Parentheses is in the wrong place.

**Comment by andrepereiradasilva** [comment]
@brianteeman please remove the RTC on thsi one to.
same parentheses problems.

**Comment by wojsmol** [comment]
@zero-24 see https://github.com/zero-24/joomla-cms/pull/14

**Comment by andrepereiradasilva** [comment]
seems fine to me now.

**Comment by zero-24** [comment]
Back to Pending. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 2f4ef513997235a7c9ffec60ed145d122cc95f6c<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 2f4ef513997235a7c9ffec60ed145d122cc95f6c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10431">issues.joomla.org/joomla-cms/10431</a>.</sub>

**Pull request merged**